src/LowLevelMotionController.cpp

```diff
@@ -231,36 +265,120 @@ int main(int argc, char **argv)
         {
             last_time = current_time;
 
+            if(server.isNewGoalAvailable() && !server.isActive())
+            {
+                const iarc7_motion::GroundInteractionGoalConstPtr& goal = server.acceptNewGoal();
+                if ("takeoff" == goal->interaction_type)
+                {
+                    bool success = takeoffController.resetForTakeover(current_time);
+                    if(success)
+                    {
+                        ROS_DEBUG("Transitioning to takeoff mode");
+                        motion_state = MotionState::TAKEOFF;
+                    }
+                    else
+                    {
+                        ROS_ERROR("Failure transitioning to takeoff mode");
+                        server.setAborted();
+                        motion_state = MotionState::VELOCITY_CONTROL;
```

Comment:
We go back to velocity mode if there is a failure to transition into takeoff mode. It's the one case were it seemed safe to allow fault recovery of some kind since the handoff between controller types has not occurred yet.

I like the idea of a GROUNDED mode. I think it would make more sense in another PR though.